src/plugins/data/common/es_query/kuery/ast/kuery.peg

```diff
@@ -28,15 +28,15 @@ start
 OrQuery
   = &{ return errorOnLuceneSyntax; } LuceneQuery
   / left:AndQuery Or right:OrQuery {
-    const cursor = [left, right].find(node => node.type === 'cursor');
+    const cursor = parseCursor && [left, right].find(node => node.type === 'cursor');
```

Comment:
`parseCursor` is an option passed to the parser which essentially specifies whether we're parsing for autocomplete suggestions or not. If `false`, then we can short-circuit any autocomplete logic (stuff where `node.type === 'cursor'`).